api/src/services/SolrIndexer.ts

```diff
@@ -281,14 +275,26 @@ class SolrIndexer {
                 fields.collection_title_sort_str = fields.title_sort;
             }
         }
-
-        fields.has_order_str = ((fields["relsext.sortOn_txt_mv"] ?? [])[0] ?? "title") === "custom" ? "yes" : "no";
-
+        // Fedora 3 stored some data in Fedora object XML and some in separate RELS-EXT datastreams.
+        // Legacy VuDL indexed these two data sources using different prefixes. For stability of legacy
+        // queries, we retain this prefix separation, using the table below to identify the former
+        // RELS-EXT fields and defaulting to the "fgs." prefix for everything else.
+        const prefixes = {
```

Comment:
you can maybe reverse this mapping to do "relsext => ["hasLegacyUrl", "hasModal", ...] "

Reply:
I used this mapping to keep the logic simple. If I reverse the map, then the logic gets more complicated. I took a stab at it to see what it would look like, but it's actually one line longer this way:

<pre>
         // queries, we retain this prefix separation, using the table below to identify the former
         // RELS-EXT fields and defaulting to the "fgs." prefix for everything else.
         const prefixes = {
-            hasLegacyURL: "relsext",
-            hasModel: "relsext",
-            itemID: "relsext",
-            isMemberOf: "relsext",
-            sequence: "relsext",
-            sortOn: "relsext",
+            relsext: ["hasLegacyURL", "hasModel", "itemID", "isMemberOf", "sequence", "sortOn"],
         };
         for (const field in fedoraData.fedoraDetails) {
-            const prefix = prefixes[field] ?? "fgs";
+            let prefix = "fgs";
+            for (const key in prefixes) {
+                if (prefixes[key].indexOf(field) !== -1) {
+                    prefix = key;
+                }
+            };
             const fieldName = prefix + "." + field + "_txt_mv";
             fields[fieldName] = fedoraData.fedoraDetails[field];
         }
</pre>

I'm fine with committing this version if you like it better... or if you have suggestions for making it more efficient!

Reply:
okay.